srtcore/buffer_tools.h

```diff
@@ -148,9 +148,87 @@ class CSndRateEstimator
     /// including the current sampling interval.
     int getCurrentRate() const;
 
+protected:
+    time_point m_tsFirstSampleTime; //< Start time of the first sample.
+    int        m_iCurSampleIdx;     //< Index of the current sample being collected.
+    int        m_iRateBps;          //< Rate in Bytes/sec.
+
+    struct Sample
+    {
+        int m_iPktsCount;  // number of payload packets
+        int m_iBytesCount; // number of payload bytes
+
+        void reset()
```

Comment:
I would also suggest to keep the CSndRateEstimator as it was to minimize the scope of changes.